rnd/autogpt_server/autogpt_server/data/block.py

```diff
@@ -0,0 +1,130 @@
+import inspect
+import json
+
+from abc import ABC, abstractmethod
+from prisma.models import AgentBlock
+from pydantic import BaseModel
+from typing import ClassVar
+
+
+class Block(ABC, BaseModel):
+    @property
+    @abstractmethod
+    def input_schema(self) -> dict[str, str]:
+        """
+        The schema for the block input data. The keys are the names of the input,
+        and the values are the types of the input.
+        Example:
+        {
+            "system_prompt": "str",
+            "user_prompt": "str",
+            "max_tokens": "int",
+        }
+        """
+        pass
+
+    @property
+    @abstractmethod
+    def output_schema(self) -> dict[str, str]:
+        """
+        The schema for the block possible output. The keys are the names of the output,
+        and the values are the types of the output.
+        Example:
+        {
+            "on_completion": "str",
+            "on_failure": "str",
+        }
+        """
+        pass
```

Comment:
This looks like only a single value for each output, how about multiple values? The same applies if we want multiple inputs. With current schema for example: `list[dict[str, str]]`.

Reply:
you can set `list[dict[str, str]]` as an output. e.g:
```
{
    "output": "list[dict[str, str]]",
}
```
Though the block will only produce 1 output at a time.